nova/virt/vmwareapi/vmops.py

```diff
@@ -2029,6 +2066,49 @@ def finish_migration(self, context, migration, instance, disk_info,
                                        step=6,
                                        total_steps=RESIZE_TOTAL_STEPS)
 
+    def _relocate_vm(self, vm_ref, context, instance, network_info,
+                     image_meta=None):
+        image_meta = image_meta or instance.image_meta
+        image_info = images.VMwareImage.from_image(context, instance.image_ref,
+                                                   image_meta)
+        extra_specs = self._get_extra_specs(instance.flavor, image_meta)
+        allowed_ds_types = ds_util.get_allowed_datastore_types(
+            image_info.disk_type)
+        datastore = ds_util.get_datastore(self._session, self._cluster,
+                                          self._datastore_regex,
+                                          extra_specs.storage_policy,
+                                          allowed_ds_types)
+        dc_info = self.get_datacenter_ref_and_name(datastore.ref)
+        folder = self._get_project_folder(dc_info, instance.project_id,
+                                          'Instances')
+
+        spec = vm_util.relocate_vm_spec(self._session.vim.client.factory,
+                                            res_pool=self._root_resource_pool,
+                                            folder=folder, datastore=datastore)
+        if network_info:
+            image_info = images.VMwareImage.from_image(context,
+                                                       instance.image_ref,
+                                                       image_meta)
```

Comment:
You already have this at the beginning of the function. I don't think you need to create it again, do you?